### DISCUSSION TITLE
Merge without waiting for requirements to be met (bypass branch protections)

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#24537] [Suggest an Idea]
### Tell us more.

GitHub provides an option to bypass the branch protections.  
It would be valuable if renovate would be able to merge the PR when the following conditions are met:
- automerge enabled
- The user which runs renovate has the permission to bypass branch protections

**Comment by viceice** [comment]
this should probably work if you use platform automerge?

**Reply by lmilbaum** [reply]
What is "platform automerge"? Can you refer to the documentation? I didn't find it.

**Reply by rarkins** [reply]
`platformAutomerge`

**Reply by HonkingGoose** [reply]
I find the idea of "let Renovate automerge past the branch protections" to be a bit silly. I would only want a bot to automerge things if my tests and branch protection rules are good to go.

If you want to bypass the branch protection rules, why not have a human manually merge the PR?

**Reply by lmilbaum** [reply]
I understand why you find the idea to be a bit silly.  There is a gentle balance between enforcing the branch protection rules and renovate actions. An organization with 10x repos renovate enabled can potentially produce 10x PRs per week if not more then that. Manually merging those can potentially break this gentle balance which would result in negative DevEx.  If I am asked, I would expect human crafted PRs to obey the branch protection rules while bot crafted ones not so much.  Does that answer your question?

**Reply by viceice** [reply]
if renovate bypass the protection it will probably break your target branch

**Reply by lmilbaum** [reply]
Could you please clarify why it will probably break my target branch?
I just want to bypass the required reviewal policy.

**Reply by viceice** [reply]
doesn't mean bypass on GitHub means bypass all branch protection? required status checks are there too.

why not just use the renovate automerge GitHub apps to get the required approval? we do this on this repo too.

**Reply by rarkins** [reply]
https://github.com/apps/renovate-approve

**Reply by lmilbaum** [reply]
One of my colleagues figured this out. I thought it would be helpful to share it here.
Updating the `Allow specified actors to bypass required pull requests` policy on the branch protection works for us.

**Comment by lmilbaum** [comment]
I've [used](https://github.com/platform-engineering-org/helper/blob/main/renovate.json) the `platformAutomerge` key as suggested with the same result. The [PR](https://github.com/platform-engineering-org/helper/pull/83) is not merged.

**Reply by rarkins** [reply]
platformAutomerge is only set at PR creation time. That PR is 14 hours old while you only made the change an hour ago. So you need to either wait for the next automerging PR or you need to close and rename the current PR so that it's recreated by the bot

**Reply by lmilbaum** [reply]
New PR... same results  https://github.com/platform-engineering-org/helper/pull/85

**Reply by rarkins** [reply]
Please paste in here the branch logs when the PR was created

**Reply by lmilbaum** [reply]
https://github.com/platform-engineering-org/helper/pull/87

```
DEBUG: Creating PR (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
       "prTitle": "chore(deps): update patch updates"
DEBUG: Creating PR (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
       "title": "chore(deps): update patch updates",
       "head": "platform-engineering-org:renovate/patch-updates",
       "base": "main",
       "draft": false
DEBUG: PR created (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
       "pr": 87,
       "draft": false
DEBUG: Adding labels '' to #87 (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: GitHub-native automerge: success...PrNo: 87 (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
 INFO: PR created (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
       "pr": 87,
       "prTitle": "chore(deps): update patch updates"
DEBUG: resolveBranchStatus(branchName=renovate/patch-updates, ignoreTests=false) (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: getBranchStatus(renovate/patch-updates) (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: branch status check result (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
       "state": "pending",
       "statuses": []
DEBUG: Successful checks are all internal renovate/ checks, so returning "pending" branch status (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: No check runs found (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
       "result": {"total_count": 0, "check_runs": []}
DEBUG: Branch status yellow (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: Skipping assignees and reviewers as automerge=true (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: setPrCache() (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: Created Pull Request #87 (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: PR is configured for automerge (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
DEBUG: setBranchCommit() (repository=platform-engineering-org/helper, branch=renovate/patch-updates)
```
